Lib/idlelib/idle_test/test_configdialog.py

```diff
@@ -568,6 +689,17 @@ def test_create_new_and_save_new(self):
         self.assertFalse(d.theme_source.get())  # Use custom set.
         eq(d.set_theme_type.called, 1)
 
+        # Call with same name as first time to make sure it adds it only once.
```

Comment:
I presume the coverage aim here is to call save_new so that the condition in
```
        if not idleConf.userCfg['highlight'].has_section(theme_name): 
            idleConf.userCfg['highlight'].add_section(theme_name) 
```
is false.    It turns out that this new test block is not needed for 2 reasons. 1. user configs have a method embodying this logic (which is now being used):
```
    def AddSection(self, section):
        "If section doesn't exist, add it."
        if not self.has_section(section):
            self.add_section(section)
```
2. The current section name dialog (which I added in about 2015 or later) gets a list of existing section names and will not return any of those as new.  So save_new should never be called with a duplicate name.  I don't know what happened before.

The same applies to new keyset names.

Reply:
Makes sense.  I think there was a comment at the time to make sure there weren't duplicates.  It may have been for a different part of the code and I tested it here just to make sure, but I don't recall.